src/python/grpcio/grpc/_channel.py

```diff
@@ -212,7 +214,14 @@ def consume_request_iterator():
         )
         call.start_batch(cygrpc.Operations(operations), event_handler)
         state.due.add(cygrpc.OperationType.send_close_from_client)
-  thread = threading.Thread(target=consume_request_iterator)
+
+  def stop_write_thread(timeout):
+    write_thread_stopped = True
```

Comment:
Good catch, and I agree.  I've also switched the tests to use infinite request iterators to catch this stuff.